src/Workspaces/Core/Portable/Workspace/Solution/DocumentInfo.cs

```diff
@@ -234,7 +256,33 @@ Checksum IChecksummedObject.Checksum
                 {
                     if (_lazyChecksum == null)
                     {
-                        _lazyChecksum = Checksum.Create(nameof(DocumentAttributes), this);
+                        using (var stream = SerializableBytes.CreateWritableStream())
+                        using (var writer = new ObjectWriter(stream))
+                        {
+                            writer.WriteString(nameof(DocumentAttributes));
+
+                            if (FilePath == null || ProjectFilePath == null)
+                            {
+                                // this checksum is not persistable because
+                                // this info doesn't have non volatile info
+                                Id.WriteTo(writer);
```

Comment:
solution not built from file can have only id but not file path, think solution we create in unit test or forked solution and etc.